ros/src/aero_offboard_velocity/src/aero_drone_offboard.cpp

```diff
@@ -0,0 +1,184 @@
+  land_client_ = nh_.serviceClient<mavros_msgs::CommandTOL>("/mavros/cmd/land");
+}
+
+AeroDrone::~AeroDrone()
+{
+}
+
+bool AeroDrone::arm()
+{
+  if (!home_set_)
+  {
+    ROS_ERROR("Can't arm: No GPS Fix!");
+    return false;
+  }
+
+  auto arming_client = nh_.serviceClient<mavros_msgs::CommandBool>("/mavros/cmd/arming");
+  mavros_msgs::CommandBool srv_arm;
+  srv_arm.request.value = true;
+  if (arming_client.call(srv_arm) && srv_arm.response.success)
+    return true;
+  else
+    return false;
+}
+
+bool AeroDrone::takeoff()
+{
+  if (!home_set_)
+  {
+    ROS_ERROR("Can't takeoff: No GPS Fix!");
+    return false;
+  }
+
+  mavros_msgs::CommandTOL srv_takeoff{};
+  srv_takeoff.request.altitude = altitude_in_amsl_;
+  srv_takeoff.request.latitude = home_.geo.latitude;
+  srv_takeoff.request.longitude = home_.geo.longitude;
+
+  if (takeoff_client_.call(srv_takeoff) && srv_takeoff.response.success)
+    return true;
+  else
+    return false;
+}
+
+bool AeroDrone::land()
+{
+  if (!home_set_)
+  {
+    ROS_ERROR("Can't land: No GPS Fix!");
+    return false;
+  }
+
+  mavros_msgs::CommandTOL srv_land{};
+
+  if (land_client_.call(srv_land) && srv_land.response.success)
+    return true;
+  else
+    return false;
+}
+
+bool AeroDrone::setOffboardMode()
+{
+  mavros_msgs::SetMode offb_set_mode;
+  offb_set_mode.request.custom_mode = "OFFBOARD";
+  if (set_mode_client_.call(offb_set_mode) && offb_set_mode.response.mode_sent)
+    return true;
+  else
+    return false;
+}
+
+void AeroDrone::getHomeGeoPoint()
+{
+  // FCU Home position: See http://docs.ros.org/api/sensor_msgs/html/msg/NavSatFix.html
+  auto home_sub = nh_.subscribe<mavros_msgs::HomePosition>("mavros/home_position/home", 1,
+                                                           boost::bind(&AeroDrone::setHomeGeoPointCB, this, _1));
+
+  ROS_INFO("Waiting for Aero FC Home to be set...");
+  while (ros::ok() && !home_set_)
+  {
+    ros::spinOnce();
+    rate_.sleep();
+  }
+}
+
+// Callback that gets called periodically from MAVROS notifying Global Poistion of Aero FCU
+void AeroDrone::setHomeGeoPointCB(const mavros_msgs::HomePositionConstPtr& home)
+{
+  home_ = *home;
+  home_set_ = true;
+  ROS_INFO("Received Home (WGS84 datum): %lf, %lf, %lf", home_.geo.latitude, home_.geo.longitude, home_.geo.altitude);
+}
+
+void AeroDrone::getAltitude()
+{
+  ros::Subscriber altitude_sub =
+      nh_.subscribe<mavros_msgs::Altitude>("/mavros/altitude", 1, boost::bind(&AeroDrone::getAltitudeCB, this, _1));
+  while (ros::ok() && !altitude_received_)
+  {
+    ros::spinOnce();
+    rate_.sleep();
+  }
+}
+
+void AeroDrone::getAltitudeCB(const mavros_msgs::AltitudeConstPtr& altitude)
+{
+  altitude_ = *altitude;
+  altitude_received_ = true;
+  altitude_in_amsl_ = altitude_.amsl;
+}
+
+float AeroDrone::toRadFromDeg(float deg)
+{
+  return static_cast<float>(deg / 180.0f * M_PI);
+}
+
+void AeroDrone::setOffboardVelocityBody(float vx, float vy, float vz, float yaw_rate)
```

Comment:
I think we don't need to have two different functions just for 'count' parameter.

Reply:
OK. We'd two different methods to ease developer usage (every time passing 1 is mandatory for setting single offboard command.
However, we can use default arguments.
@padmashree9110 you can use as below:
```
void
AeroDrone::setOffboardVelocityBody(float vx, float vy, float vz,
                                   float yaw_rate,
                                   size_t count = 1);
// count 1 is passed by default; can be overridden
```